src/nomad_simulations/schema_packages/model_method.py

```diff
@@ -1221,3 +1221,259 @@ class DMFT(ModelMethodElectronic):
+        total_occupation = sum(
+            orbital.occupation
+            for orbital in self.molecular_orbitals
+            if orbital.occupation is not None
+        )
+        expected_occupation = self.n_alpha_electrons + self.n_beta_electrons
+        if total_occupation != expected_occupation:
+            logger.warning(
+                f'The total occupation ({total_occupation}) does not match the expected value ({expected_occupation}).'
+            )
+
+
+class GTOIntegralDecomposition(BaseModelMethod):
+    """
+    A general class for integral decomposition techniques for Coulomb and exchange integrals.
+    Examples:
+    Resolution of identity (RI-approximation):
+    RI
+    RIJK
+    ....
+    Chain-of-spheres (COSX) algorithm for exchange: doi:10.1016/j.chemphys.2008.10.036
+    """
+
+    approximation_type = Quantity(
+        type=str,
+        description="""
+        RIJ, RIK, RIJK,
+        """,
+    )
+
+    approximated_term = Quantity(
+        type=str,
+        description="""
+        such as coulomb, exchange, explicit-correlation
+        to be converted to an MEnum later.
+        """,
+    )
+
+    def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
+        super().normalize(archive, logger)
+
+
+class HartreeFock(ModelMethodElectronic):
+    """
+    A base section for Hartree Fock ansatz.
+    """
+
+    reference_determinant = Quantity(
+        type=MEnum('UHF', 'RHF', 'ROHF', 'UKS', 'RKS', 'ROKS'),
+        description="""
+        the type of reference determinant.
+        """,
+    )
+
+
+class PerturbationMethod(ModelMethodElectronic):
+    type = Quantity(
+        type=MEnum('MP', 'RS', 'BW'),
+        description="""
+        Perturbation approach. The abbreviations stand for:
+        | Abbreviation | Description |
+        | ------------ | ----------- |
+        | `'MP'`       | Moller-Plesset |
+        | `'RS'`       | Rayleigh-Schrödigner |
+        | `'BW'`       | Brillouin-Wigner |
+        """,
+        a_eln=ELNAnnotation(component='EnumEditQuantity'),
+    )  # TODO: check if the special symbols are supported
+
+    order = Quantity(
+        type=np.int32,
+        description="""
+        Order up to which the perturbation is expanded.
+        """,
+        a_eln=ELNAnnotation(component='NumberEditQuantity'),
+    )
+
+    density = Quantity(
+        type=MEnum('relaxed', 'unrelaxed'),
+        description="""
+        unrelaxed density: MP2 expectation value density
+        relaxed density  : incorporates orbital relaxation
+        """,
+    )
+
+
+class LocalCorrelation(ArchiveSection):
+    """
+    A base section used to define the parameters of a local correlation for the post-HF methods.
+
+    It has a corresponding localization method.
+    """
+
+    method = Quantity(
+        type=MEnum('LMP2', 'LCCD', 'LCCSD', 'LCCSD(T)', 'DLPNO-CCSD(T)', 'LocalDFT'),
+        description="""
+        The local correlation method applied. For example:
```

Comment:
Pls elaborate a bit more here. Feel free to also add references for specific methods.